### PR TITLE
Spectrogram template-ing

### PR DESCRIPTION
**Pull request by birdsarah**
handles "use new embed.components to have better layout design" on #2433

### CODE REVIEW

**Comment by birdsarah** [comment]
This is ready for review.

File: examples/embed/spectrogram/spectrogram.py [hunk 1]
```diff
@@ -171,14 +156,39 @@ def make_spectrogram():
         inner_radius="inner_radius", outer_radius="outer_radius",
         start_angle="start_angle", end_angle="end_angle",
         source=radial_source, name="eq")
-    eq.grid.grid_line_color=None
+    eq.grid.grid_line_color = None
+
+    return freq_slider, gain_slider, spectrum, signal, spec, eq
+
+
+def get_layout():
```
Comment (bryevdv) [inline examples/embed/spectrogram/spectrogram.py]: This is unused, right? Let's just delete it, I think leaving it would be confusing.

**Comment by birdsarah** [comment]
@bryevdv I've removed get_layout that I left lurking around.

File: examples/embed/spectrogram/templates/spectrogram.html [hunk 2]
```diff
@@ -5,13 +5,64 @@
...
+
+    {# end CONTENT BLOCK #}
+
+    {# start JS BLOCK #}
+
+        {% for raw in bokeh.js_raw %}
+            <script type="text/javascript">{{ raw|safe }}</script>
+        {% endfor %}
+
+        {{ script|indent(8)|safe }}
```
Comment (damianavila) [inline examples/embed/spectrogram/templates/spectrogram.html]: curious... why indent (8)?

File: examples/embed/spectrogram/templates/spectrogram.html [hunk 3]
```diff
@@ -5,13 +5,64 @@
...
+        <script type="text/javascript" src="static/js/spectrogram.js" defer></script>
```
Comment (damianavila) [inline examples/embed/spectrogram/templates/spectrogram.html]: defer, I did not know about it... thanks!

**Comment by damianavila** [comment]
LGTM, but I could not test it because some problem in my ALSA library (pyaudio is always messy in this machine... I will try later to dig in the issue so I can test it).

File: examples/embed/spectrogram/templates/spectrogram.html (hunk 2 above)
Comment (birdsarah) [inline examples/embed/spectrogram/templates/spectrogram.html]: we seem to do it elsewhere.......(aka no good reason i am a :sheep:)

File: examples/embed/spectrogram/templates/spectrogram.html (hunk 3 above)
Comment (birdsarah) [inline examples/embed/spectrogram/templates/spectrogram.html]: The defer was already there I just copied and pasted. I'm almost certain it's not necessary as this is the last block in the page.

Comment (damianavila) [inline examples/embed/spectrogram/templates/spectrogram.html]: > The defer was already there I just copied and pasted

Well then I overlooked before...

>  I'm almost certain it's not necessary as this is the last block in the page.

Yep, I agree...

**Comment by damianavila** [comment]
@bryevdv, I need your input here... since LGTM, I will merge this one in a little bit if you don't stop me...

**Comment by damianavila** [comment]
OK, merging this one now...

**Pull request merged**